### PR TITLE
Support for dynamic query parameters and method overloads

### PR DESCRIPTION
**Pull request by jlucansky**

### CODE REVIEW

**Comment by bennor** [comment]
A couple of things: 

1. Can you give an example of the proposed dynamic query parameters in action, or better, add some docs to the readme for it.
2. Why use an attribute for distinguishing overloads when you could just use the parameter types to determine uniqueness. Seems like an extra layer of complexity we could do without. Ideally overloads would _just work_.

**Comment by onovotny** [comment]
Can you please rebase and ensure it builds correctly? Some of the added files seem to be missing from some of the projects.

Thanks!

File: Refit/RequestBuilderImplementation.cs [hunk 1]
```diff
@@ -446,6 +463,13 @@ public RestMethodInfo(Type targetInterface, MethodInfo methodInfo, RefitSettings
             Name = methodInfo.Name;
             MethodInfo = methodInfo;
 
+            var uma = methodInfo.GetCustomAttributes(true)
+                .OfType<UniqueNameAttribute>()
+                .FirstOrDefault();
+
+            if (uma != null)
+                Name = uma.MethodName;
```
Comment (ahmedalejo) [inline Refit/RequestBuilderImplementation.cs]: `UniqueNameAttribute` should not be compulsory for supporting overloads
name should default to method mangling #85 or [#82 Comment](https://github.com/paulcbetts/refit/issues/82#issuecomment-66140879)

Comment (jlucansky) [inline Refit/RequestBuilderImplementation.cs]: Yes you are right and I wish there was another way, but it is almost impossible to get the same method signature in runtime as during stub generation.

**Comment by onovotny** [comment]
Closing due to age. Refit also now supports method overloads

**Comment by bennor** [comment]
Both of these features exist. Overloads should _just work_. Dynamic query parameters are supported via the `[Query]` attribute.

**Pull request closed without merging**